eureka-core/src/main/java/com/netflix/eureka/registry/AwsInstanceRegistry.java

```diff
@@ -0,0 +1,122 @@
+/*
+ * Copyright 2015 Netflix, Inc.
+ *
+ *    Licensed under the Apache License, Version 2.0 (the "License");
+ *    you may not use this file except in compliance with the License.
+ *    You may obtain a copy of the License at
+ *
+ *        http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *    Unless required by applicable law or agreed to in writing, software
+ *    distributed under the License is distributed on an "AS IS" BASIS,
+ *    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *    See the License for the specific language governing permissions and
+ *    limitations under the License.
+ */
+
+package com.netflix.eureka.registry;
+
+import com.netflix.appinfo.InstanceInfo;
+import com.netflix.discovery.EurekaClient;
+import com.netflix.discovery.EurekaClientConfig;
+import com.netflix.eureka.EurekaServerConfig;
+import com.netflix.eureka.aws.AwsAsgUtil;
+import com.netflix.eureka.cluster.PeerEurekaNodes;
+import com.netflix.eureka.lease.Lease;
+import com.netflix.eureka.resources.ServerCodecs;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import javax.inject.Inject;
+import javax.inject.Singleton;
+
+/**
+ * Override some methods with aws specific usecases.
+ *
+ * @author David Liu
+ */
+@Singleton
+public class AwsInstanceRegistry extends PeerAwareInstanceRegistryImpl {
+    private static final Logger logger = LoggerFactory.getLogger(AwsInstanceRegistry.class);
+
+    private AwsAsgUtil awsAsgUtil;
+
+    @Inject
+    public AwsInstanceRegistry(EurekaServerConfig serverConfig,
+                               EurekaClientConfig clientConfig,
+                               ServerCodecs serverCodecs,
+                               EurekaClient eurekaClient) {
+        super(serverConfig, clientConfig, serverCodecs, eurekaClient);
+    }
+
+    @Override
+    public void init(PeerEurekaNodes peerEurekaNodes) throws Exception {
+        super.init(peerEurekaNodes);
+        this.awsAsgUtil = new AwsAsgUtil(serverConfig, clientConfig, this);
```

Comment:
It is no longer util. Can we convert this into a regular service with an interface and implementation, and inject it via constructor?